### PR TITLE
add and update carrier endpoints

### PR DESCRIPTION
**Pull request by adnaniroshanak**

### CODE REVIEW

**Review by MarkusWallnerFul — APPROVED** [review]

File: api.swagger.yaml [hunk 1]
```diff
@@ -1706,6 +1706,91 @@ paths:
...
+      parameters:
+        - in: "body"
+          name: "body"
+          description: "Carrier object"
+          required: true
+          schema:
+            $ref: "#/definitions/CarrierForCreation"
+      responses:
+        201:
+          description: "The carrier was successfully created. The Location header contains the URL of the parcel."
```
Comment (timdauer) [inline api.swagger.yaml:1735]:
```suggestion
          description: "The carrier was successfully created. The Location header contains the URL of the carrier."
```

File: api.swagger.yaml [hunk 2]
```diff
@@ -1706,6 +1706,91 @@ paths:
...
+          schema:
+            $ref: "#/definitions/CarrierCredential"
```
Comment (timdauer) [inline api.swagger.yaml:1737]:
```suggestion
            $ref: "#/definitions/Carrier"
```

File: api.swagger.yaml [hunk 3]
```diff
@@ -3338,18 +3420,58 @@ definitions:
...
+        type: "object"
+        properties:
+          key:
+            type: "string"
+            enum:
+              - "DHL"
+            description: "Use value 'DHL' in order to provide DHL credentials"
+          dhlGkUsername:
+            type: "string"
+            description: "Username you use to login to the DHL Geschäftskundenportal"
```
Comment (timdauer) [inline api.swagger.yaml:3466]:
```suggestion
            description: "Username (Benutzername) you use to login to the DHL Geschäftskundenportal"
```

File: api.swagger.yaml [hunk 4]
```diff
@@ -3338,18 +3420,58 @@ definitions:
...
+          dhlGKPassword:
+            type: "string"
+            description: "Password you use to login to the DHL Geschäftskundenportal"
+          dhlEkpNumber:
```
Comment (timdauer) [inline api.swagger.yaml:3470]:
```suggestion
          dhlAccountNumber:
```

File: api.swagger.yaml [hunk 5]
```diff
@@ -3338,18 +3420,58 @@ definitions:
...
+            type: "string"
+            description: "EKP number used within to the DHL Geschäftskundenportal"
```
Comment (timdauer) [inline api.swagger.yaml:3472]:
```suggestion
            description: "Abrechnungsnummer DHL Paket, displayed in the DHL Geschäftskundenportal under 'Vertragspositionen'"
```

File: api.swagger.yaml [hunk 6]
```diff
@@ -3502,3 +3624,17 @@ definitions:
         additionalProperties: false
     xml:
       name: "ModifyHandoverjobAction"
+  CarrierCredential:
+    allOf:
+      - $ref: '#/definitions/CarrierForCreation'
+      - $ref: '#/definitions/VersionedResource'
```
Comment (timdauer) [inline api.swagger.yaml:3630]:
```suggestion

```

File: api.swagger.yaml [hunk 7]
```diff
@@ -3502,3 +3624,17 @@ definitions:
         additionalProperties: false
     xml:
       name: "ModifyHandoverjobAction"
+  CarrierCredential:
```
Comment (timdauer) [inline api.swagger.yaml:3627]:
```suggestion
  Carrier:
```

File: api.swagger.yaml [hunk 8]
```diff
@@ -3502,3 +3624,17 @@ definitions:
...
+      - $ref: '#/definitions/CarrierForCreation'
+      - $ref: '#/definitions/VersionedResource'
+      - required:
+          - "id"
+        type: "object"
+        properties:
+          id:
+            type: "string"
+            example: "LGMl2DuvPnfPoSHhYFOm"
+            description: "The id of the carrier credential"
```
Comment (timdauer) [inline api.swagger.yaml:3638]:
```suggestion
            description: "The id of the carrier"
```

**Review by thees — APPROVED** [review]

File: api.swagger.yaml [hunk 9]
```diff
@@ -3338,18 +3420,58 @@ definitions:
...
+        description: "Use value 'DHL' when you want to create DHL carrier "
+    xml:
+      name: "AbstractCarrierCredential"
+  DhlCarrierCredential:
+    allOf:
+      - $ref: '#/definitions/AbstractCarrierCredential'
+      - required:
+          - "dhlGkUsername"
+          - "dhlGKPassword"
+          - "dhlEkpNumber"
```
Comment (timdauer) [inline api.swagger.yaml:3455]:
```suggestion
          - "dhlAccountNumber"
```

**Review by timdauer — CHANGES_REQUESTED** [review]

File: api.swagger.yaml [hunk 10]
```diff
@@ -2314,12 +2399,9 @@ definitions:
         description: "This value gives the priority in the respective attribute category. The lower the value the higher is the priority, e.g. priority 1 is higher than priority 10. Attributes that have the highest priority might be selected for display in different articles of OCFF. Default Value is 1001. For details please contact the product owners."
       key:
         type: "string"
-        minLength: 1
-        example: "%%subtitle%%"
-        description: "Providing the key %%subtitle%% (see example) here will cause the value to appear for example in the App directly under the title. With all other attributes also the key will be displayed in the clients."
+        example: "subtitle"
```
Comment (timdauer) [inline api.swagger.yaml:2402]:
```suggestion
        example: "%%subtitle%%"
```

File: api.swagger.yaml [hunk 11]
```diff
@@ -2314,12 +2399,9 @@ definitions:
         description: "This value gives the priority in the respective attribute category. The lower the value the higher is the priority, e.g. priority 1 is higher than priority 10. Attributes that have the highest priority might be selected for display in different articles of OCFF. Default Value is 1001. For details please contact the product owners."
       key:
         type: "string"
-        minLength: 1
```
Comment (timdauer) [inline api.swagger.yaml:2317]: The following three attributes must stay (minLengt, example & description) as they were

**Review by timdauer — CHANGES_REQUESTED** [review]

**Pull request merged**